lib/openshift4-monitoring-operator-rules.libsonnet

```
@@ -0,0 +1,190 @@
+    name: global_name,
+  },
+  subjects: [
+    {
+      kind: 'ServiceAccount',
+      name: serviceAccount(ns).metadata.name,
+      namespace: ns,
+    },
+  ],
+};
+
+/**
+ * \brief create a minimal ManagedResource for managing operator PrometheusRules.
+ *
+ * \arg ns
+ *        The namespace to deploy the ManagedResource to.
+ * \returns
+ *        A ManagedResource CR for managing operator PrometheusRules.
+ */
+local managedResource(ns) = esp.managedResource(global_name, ns) {
+  spec: {
+    serviceAccountRef: { name: serviceAccount(ns).metadata.name },
+    context: [
+      {
+        name: 'op_rules',
+        resource: {
+          apiVersion: 'monitoring.coreos.com/v1',
+          kind: 'PrometheusRule',
+          labelSelector: {
+            // Only match rules that are not created by espejote: 'espejote.io/created-by'
+            // and are not marked to be ignored: 'espejote.io/ignore'
+            matchExpressions: [
+              {
+                key: 'espejote.io/created-by',
+                operator: 'DoesNotExist',
+              },
+              {
+                key: 'espejote.io/ignore',
+                operator: 'DoesNotExist',
+              },
+            ],
+          },
+        },
+      },
+    ],
+    triggers: [
+      {
+        name: 'op_rules',
+        watchContextResource: {
+          name: 'op_rules',
+        },
+      },
+      {
+        name: 'generated_rules',
+        watchResource: {
+          apiVersion: 'monitoring.coreos.com/v1',
+          kind: 'PrometheusRule',
+          labelSelector: {
+            // Only match rules that are created by espejote: 'espejote.io/created-by=openshift4-monitoring-rules'
+            matchExpressions: [
+              {
+                key: 'espejote.io/created-by',
+                operator: 'In',
+                values: [ 'openshift4-monitoring-rules' ],
+              },
+            ],
+          },
+        },
+      },
+      //   {
+      //     name: 'jslib_global',
+      //     watchResource: {
+      //       apiVersion: esp_gv,
+      //       kind: 'JsonnetLibrary',
+      //       name: global_name,
+      //       namespace: global_namespace,
+      //     },
+      //   },
+      //   {
+      //     name: 'jslib_component',
+      //     watchResource: {
+      //       apiVersion: jsonnetLibrary(ns).apiVersion,
+      //       kind: jsonnetLibrary(ns).kind,
+      //       name: jsonnetLibrary(ns).metadata.name,
+      //     },
+      //   },
+    ],
+    template: |||
+      local esp = import 'espejote.libsonnet';
+
+      local renderer = import 'lib/openshift4-monitoring-rules/renderer_v1.libsonnet';
+      local configGlobal = import 'lib/openshift4-monitoring-rules/config_v1.json';
+      local configComponent = import 'openshift4-monitoring-rules/config.json';
+
+      local opRules = esp.context().op_rules;
+
+      [
+        renderer.process(or, configGlobal, configComponent),
+        for or in opRules
+      ]
```

we probably want to skip rules that have a deletion timestamp:

```suggestion
      [
        renderer.process(or, configGlobal, configComponent),
        for or in opRules
        if std.get(or.metadata, 'deletionTimestamp') != null
      ]
```